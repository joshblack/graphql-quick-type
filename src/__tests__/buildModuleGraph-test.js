'use strict';

const {types: t, isPrimitiveType} = require('../types');
const areEqual = require('fbjs/lib/areEqual');
const traverse = require('../traverse');
const inferType = require('../inferType');
const buildModuleGraph = require('../buildModuleGraph');

const input = {
  id: 'a',
  name: 'Jane Doe',
  info: {
    verified: true,
  },
};

describe('buildModuleGraph', () => {
  it('should work', () => {
    expect(buildModuleGraph(input)).toMatchSnapshot();
  });
});
