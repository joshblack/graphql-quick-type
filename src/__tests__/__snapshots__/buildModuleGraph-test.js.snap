// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildModuleGraph should work 1`] = `
Array [
  Object {
    "exports": Object {
      "default": Object {
        "fields": Array [
          Object {
            "name": "id",
            "type": "GraphQLString",
          },
          Object {
            "name": "name",
            "type": "GraphQLString",
          },
          Object {
            "exports": Object {
              "default": Object {
                "fields": Array [
                  Object {
                    "name": "verified",
                    "type": "GraphQLBoolean",
                  },
                ],
                "name": "info",
                "type": "GraphQLObjectType",
              },
            },
            "id": 3,
            "imports": Array [
              Object {
                "name": "GraphQLObjectType",
                "source": "graphql",
              },
              Object {
                "name": "GraphQLBoolean",
                "source": "graphql",
              },
            ],
            "name": "generatedType3",
          },
        ],
        "type": "GraphQLObjectType",
      },
    },
    "id": 0,
    "imports": Array [
      Object {
        "name": "GraphQLObjectType",
        "source": "graphql",
      },
      Object {
        "name": "GraphQLString",
        "source": "graphql",
      },
      Object {
        "exports": Object {
          "default": Object {
            "fields": Array [
              Object {
                "name": "verified",
                "type": "GraphQLBoolean",
              },
            ],
            "name": "info",
            "type": "GraphQLObjectType",
          },
        },
        "id": 3,
        "imports": Array [
          Object {
            "name": "GraphQLObjectType",
            "source": "graphql",
          },
          Object {
            "name": "GraphQLBoolean",
            "source": "graphql",
          },
        ],
        "name": "generatedType3",
      },
    ],
    "name": "generatedType0",
  },
  Object {
    "exports": Object {
      "default": Object {
        "fields": Array [
          Object {
            "name": "verified",
            "type": "GraphQLBoolean",
          },
        ],
        "name": "info",
        "type": "GraphQLObjectType",
      },
    },
    "id": 3,
    "imports": Array [
      Object {
        "name": "GraphQLObjectType",
        "source": "graphql",
      },
      Object {
        "name": "GraphQLBoolean",
        "source": "graphql",
      },
    ],
    "name": "generatedType3",
  },
]
`;
